packages/gatsby/src/bootstrap/resolve-module-exports.ts

```diff
@@ -81,15 +81,32 @@ const staticallyAnalyzeExports = (
     // get foo from `export { foo } from 'bar'`
     // get foo from `export { foo }`
     ExportSpecifier: function ExportSpecifier(astPath) {
-      const exportName = get(astPath, `node.exported.name`)
+      const exportName = astPath?.node?.exported?.name
       isES6 = true
-      if (exportName) exportNames.push(exportName)
+      if (exportName) {
+        exportNames.push(exportName)
+      }
     },
 
     // export default () => {}
+    // export default function() {}
+    // export default function foo() {}
     // const foo = () => {}; export default foo
     ExportDefaultDeclaration: function ExportDefaultDeclaration(astPath) {
-      const name = get(astPath, `node.declaration.name`)
+      const declaration = astPath.node.declaration
+      if (
+        !t.isIdentifier(declaration) &&
+        !t.isArrowFunctionExpression(declaration) &&
+        !t.isFunctionDeclaration(declaration)
```

Comment:
Is it not relevant to check the name of a function declaration? (In that case do note a notorious exception where default exported function declarations might not have an explicit name.)

Also, it's very technically possible that this is a function expression (`export default (function(){})`), which may be less of an edge case with TS type casting or something. Is that relevant here? Those functions may or may not have an id as well.

Reply:
There are lot of things we don't account for but mosty I just want to unblock myself. I've added the function identifier to mimic arrow functions.